### PR TITLE
[Refactoring] make transformations return whether graph is modified

### PR DESCRIPTION
**Pull request by penguinwu**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#54777 [Refactoring] make transformations return whether graph is modified**

Summary: Updated RemoveListMutation, PeepholeOptimizedListIdoms,
LoopUnrolling, PeepholeOptimization to return whether graph is
modified after transformation, PeepholeAliasSensitivity

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27412105](https://our.internmc.facebook.com/intern/diff/D27412105)

### CODE REVIEW

**Comment by penguinwu** [comment]
This PR depends on #54640 which contains a fix in `jit_utils.py`. Will rebase once #54640 is merged in. Without it, there are 17 additional failures due to return type mismatch in `run_pass`

File: torch/csrc/jit/passes/loop_unrolling.cpp [hunk 1]
```diff
@@ -360,9 +365,12 @@ Node* PeelLoop(Node* n, size_t times) {
   return peeled_copy;
 }
 
-void UnrollLoops(std::shared_ptr<Graph>& graph) {
-  UnrollLoops(graph->block());
-  EliminateDeadCode(graph);
+bool UnrollLoops(std::shared_ptr<Graph>& graph) {
+  bool changed = UnrollLoops(graph->block());
+  if (changed) {
```
Comment (eellison) [inline torch/csrc/jit/passes/loop_unrolling.cpp:370]: nice

**Review by eellison — APPROVED** [review]
Nice, looks great!! thanks for doing this!! It might be nice to add a comment to the header files that the passes return whether a change was made

**Review by Krovatkin — APPROVED** [review]
:shipit:

**Pull request closed without merging**